### PR TITLE
Openshift login

### PR DESCRIPTION
**Pull request by oybed**
#### What does this PR do?
Separated out the "openshift-login" role to allow for independent use of this role

#### Which tests illustrate how this code works?
re-run the regression tests to ensure that everything works as it used to

#### Is there a relevant Issue open for this?
N/A

#### Are there any other relevant resources that should be reviewed as well?
N/A

#### Who would you like to review this?
/cc @sherl0cks @mcanoy

### CODE REVIEW

File: roles/openshift-login/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+- name: "Gather the openshift token, if not already set"
+  command: >
+    {{ openshift.common.client_binary }} whoami -t
+  register: token_output
+  when:
+  - openshift_token|trim == ''
+
+# - name: "Set openshift_token fact"
+#  set_fact:
```
Comment (sherl0cks) [inline roles/openshift-login/tasks/main.yml]: Why is this commented code here? Can it be removed?

File: roles/openshift-login/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  - auth_token|bool == True
+  - oc_login|bool == True
+
+- name: "Add token to all openshift client comands when token is set"
+  set_fact:
+    openshift:
+      common:
+        client_binary: "{{ openshift.common.client_binary }} --token={{ openshift_token }}"
+        admin_binary: "{{ openshift.common.admin_binary }} --token={{ openshift_token }}"
+  when: auth_token|bool == True
```
Comment (sherl0cks) [inline roles/openshift-login/tasks/main.yml:85]: This is an interesting way to keep the token around. I like it.

File: roles/openshift-login/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+- name: "Log in to OpenShift Client (token)"
+  command: >
+    {{ openshift.common.client_binary }} login {{ openshift_login_url }}
+    --insecure-skip-tls-verify=true --token={{ openshift_token }}
+  changed_when: False
+  when:
+  - auth_user_pass|bool == False
+  - auth_token|bool == True
+  - oc_login|bool == True
```
Comment (sherl0cks) [inline roles/openshift-login/tasks/main.yml:77]: What's the reason behind all the `bool` filters?

**Review by sherl0cks — CHANGES_REQUESTED** [review]
I'd like the commented code removed if we can

File: roles/openshift-login/tasks/main.yml (hunk 3 above)
Comment (oybed) [reply roles/openshift-login/tasks/main.yml:77]: Due to the many ansible versions we are hitting; casting it to a boolean I've found to be a bit more fail proof across versions

File: roles/openshift-login/tasks/main.yml (hunk 2 above)
Comment (oybed) [reply roles/openshift-login/tasks/main.yml:85]: Yes, this is the change Eric made. However, it doesn't work if you don't login (even though the token is valid), hence the extra login step now ...

File: roles/openshift-login/tasks/main.yml (hunk 1 above)
Comment (oybed) [reply roles/openshift-login/tasks/main.yml]: oversight ... it should be enabled, but I ran into some issues with it and went an alternative path. I'll re-visit and re-enable.

**Comment by oybed** [comment]
@sherl0cks commented out code has been fixed and is working. Please re-review and merge if you are ok with the latest.

**Review by sherl0cks — APPROVED** [review]

**Comment by sherl0cks** [comment]
lgtm

**Pull request merged**